// @nibiruchain/nibijs/dist/src/sdk/msg/devgas.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupDevgasMsgExtension = void 0;
const stargate_1 = require("@cosmjs/stargate");
const tx_1 = require("../../protojs/nibiru/devgas/v1/tx");
const setupDevgasMsgExtension = (base) => {
    const queryService = new tx_1.MsgClientImpl((0, stargate_1.createProtobufRpcClient)(base));
    return {
        devgasMsg: {
            registerFeeShare: (body) => __awaiter(void 0, void 0, void 0, function* () { return queryService.RegisterFeeShare(tx_1.MsgRegisterFeeShare.fromPartial(body)); }),
            updateFeeShare: (body) => __awaiter(void 0, void 0, void 0, function* () { return queryService.UpdateFeeShare(tx_1.MsgUpdateFeeShare.fromPartial(body)); }),
            cancelFeeShare: (body) => __awaiter(void 0, void 0, void 0, function* () { return queryService.CancelFeeShare(tx_1.MsgCancelFeeShare.fromPartial(body)); }),
            updateParams: (body) => __awaiter(void 0, void 0, void 0, function* () { return queryService.UpdateParams(tx_1.MsgUpdateParams.fromPartial(body)); }),
        },
    };
};
exports.setupDevgasMsgExtension = setupDevgasMsgExtension;
