// @nibiruchain/nibijs/dist/src/sdk/msg/oracle.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupOracleMsgExtension = void 0;
const stargate_1 = require("@cosmjs/stargate");
const tx_1 = require("../../protojs/nibiru/oracle/v1/tx");
const setupOracleMsgExtension = (base) => {
    const queryService = new tx_1.MsgClientImpl((0, stargate_1.createProtobufRpcClient)(base));
    return {
        oracleMsg: {
            aggregateExchangeRatePrevote: (body) => __awaiter(void 0, void 0, void 0, function* () {
                return queryService.AggregateExchangeRatePrevote(tx_1.MsgAggregateExchangeRatePrevote.fromPartial(body));
            }),
            aggregateExchangeRateVote: (body) => __awaiter(void 0, void 0, void 0, function* () {
                return queryService.AggregateExchangeRateVote(tx_1.MsgAggregateExchangeRateVote.fromPartial(body));
            }),
            delegateFeedConsent: (body) => __awaiter(void 0, void 0, void 0, function* () {
                return queryService.DelegateFeedConsent(tx_1.MsgDelegateFeedConsent.fromPartial(body));
            }),
            editOracleParams: (body) => __awaiter(void 0, void 0, void 0, function* () { return queryService.EditOracleParams(tx_1.MsgEditOracleParams.fromPartial(body)); }),
        },
    };
};
exports.setupOracleMsgExtension = setupOracleMsgExtension;
