// @nibiruchain/nibijs/dist/src/gql/graphql-codegen/codegen.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.changeQueryName = void 0;
const change_case_all_1 = require("change-case-all");
const changeQueryName = (str) => `GQL${(0, change_case_all_1.pascalCase)(str)}`;
exports.changeQueryName = changeQueryName;
exports.default = {
    schema: {
        codegen: {
            loader: "./src/gql/graphql-codegen/codegen-loader.js",
        },
    },
    hooks: {
        afterOneFileWrite: ["prettier --write"],
    },
    config: { namingConvention: exports.changeQueryName },
    generates: {
        "./src/gql/utils/schema.graphql": {
            plugins: ["schema-ast"],
        },
        "./src/gql/utils/generated.ts": {
            plugins: ["typescript", "typescript-operations"],
            config: {
                immutableTypes: true,
                scalars: {
                    Time: {
                        input: "string",
                        output: "string",
                    },
                },
            },
        },
    },
};
