// @nibiruchain/nibijs/dist/src/protojs/nibiru/inflation/v1/inflation.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InflationDistribution = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
function createBaseInflationDistribution() {
    return { stakingRewards: "", communityPool: "", strategicReserves: "" };
}
exports.InflationDistribution = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.stakingRewards !== "") {
            writer.uint32(10).string(message.stakingRewards);
        }
        if (message.communityPool !== "") {
            writer.uint32(18).string(message.communityPool);
        }
        if (message.strategicReserves !== "") {
            writer.uint32(26).string(message.strategicReserves);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInflationDistribution();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.stakingRewards = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.communityPool = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.strategicReserves = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            stakingRewards: isSet(object.stakingRewards) ? String(object.stakingRewards) : "",
            communityPool: isSet(object.communityPool) ? String(object.communityPool) : "",
            strategicReserves: isSet(object.strategicReserves) ? String(object.strategicReserves) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.stakingRewards !== undefined && (obj.stakingRewards = message.stakingRewards);
        message.communityPool !== undefined && (obj.communityPool = message.communityPool);
        message.strategicReserves !== undefined && (obj.strategicReserves = message.strategicReserves);
        return obj;
    },
    create(base) {
        return exports.InflationDistribution.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseInflationDistribution();
        message.stakingRewards = (_a = object.stakingRewards) !== null && _a !== void 0 ? _a : "";
        message.communityPool = (_b = object.communityPool) !== null && _b !== void 0 ? _b : "";
        message.strategicReserves = (_c = object.strategicReserves) !== null && _c !== void 0 ? _c : "";
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
