// @nibiruchain/nibijs/dist/src/gql/utils/defaultObjects.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultStakingHistoryItem = exports.defaultProxy = exports.defaultFeatureFlags = exports.defaultInflationInfo = exports.defaultInflationReward = exports.defaultInflationDistribution = exports.defaultUserContract = exports.defaultIbcChannelsResponse = exports.defaultIbc = exports.defaultIbcTransfer = exports.defaultIbcChannel = exports.defaultOracleEntry = exports.defaultOraclePrice = exports.defaultUnbondings = exports.defaultVolume = exports.defaultUsers = exports.defaultTvl = exports.defaultTotals = exports.defaultPerpPnl = exports.defaultPerpOpenInterest = exports.defaultStatsFees = exports.defaultSpotPoolSwap = exports.defaultSpotLpPosition = exports.defaultRedelegations = exports.defaultDelegations = exports.defaultDistributionCommission = exports.defaultGovernance = exports.defaultPerpLeaderboard = exports.defaultPerpPositionChanges = exports.defaultMarkPriceCandles = exports.defaultGovVote = exports.defaultGovDeposit = exports.defaultGovProposal = exports.defaultSpotPool = exports.defaultPool = exports.defaultPerpPosition = exports.defaultPerpMarket = exports.defaultUser = exports.defaultActor = exports.defaultValidator = exports.defaultBlock = exports.defaultToken = void 0;
exports.defaultToken = {
    denom: "",
    amount: "",
};
exports.defaultBlock = {
    block: 0,
    block_duration: 0,
    block_ts: 0,
    num_txs: 0,
};
exports.defaultValidator = {
    commission_update_time: "",
    commission_rates: {
        max_change_rate: 0,
        max_rate: 0,
        rate: 0,
    },
    delegator_shares: 0,
    description: {
        details: "",
        identity: "",
        moniker: "",
        security_contact: "",
        website: "",
    },
    creation_block: exports.defaultBlock,
    self_delegation: 0,
    jailed: false,
    operator_address: "",
    min_self_delegation: 0,
    status: "BONDED",
    tokens: 0,
    unbonding_block: exports.defaultBlock,
    unbonding_time: "",
};
exports.defaultActor = {
    address: "",
    balances: [exports.defaultToken],
    created_block: exports.defaultBlock,
};
exports.defaultUser = exports.defaultActor;
exports.defaultPerpMarket = {
    base_reserve: 0,
    ecosystem_fund_fee_ratio: 0,
    max_funding_rate: 0,
    enabled: true,
    exchange_fee_ratio: 0,
    funding_rate_epoch_id: "",
    index_price_twap: 0,
    is_deleted: false,
    latest_cumulative_premium_fraction: 0,
    liquidation_fee_ratio: 0,
    maintenance_margin_ratio: 0,
    mark_price: 0,
    mark_price_twap: 0,
    max_leverage: 0,
    pair: "",
    partial_liquidation_ratio: 0,
    prepaid_bad_debt: exports.defaultToken,
    price_multiplier: 0,
    quote_reserve: 0,
    sqrt_depth: 0,
    total_long: 0,
    total_short: 0,
    twap_lookback_window: "",
};
exports.defaultPerpPosition = {
    bad_debt: 0,
    last_updated_block: exports.defaultBlock,
    latest_cumulative_premium_fraction: 0,
    margin: 0,
    margin_ratio: 0,
    open_notional: 0,
    pair: "",
    liquidation_price: 0,
    position_notional: 0,
    size: 0,
    trader_address: "",
    unrealized_funding_payment: 0,
    unrealized_pnl: 0,
};
exports.defaultPool = {
    amplification: 0,
    created_block: exports.defaultBlock,
    exit_fee: 0,
    swap_fee: 0,
    pool_id: 0,
    tokens: [exports.defaultToken],
    pool_type: "",
    total_shares: exports.defaultToken,
    total_weight: 0,
    weights: [exports.defaultToken],
};
exports.defaultSpotPool = {
    block: exports.defaultBlock,
    pool: exports.defaultPool,
    pool_shares: exports.defaultToken,
    user: exports.defaultUser,
};
exports.defaultGovProposal = {
    depositEndTime: "",
    finalTallyResultAbstain: 0,
    finalTallyResultNo: 0,
    finalTallyResultNoWithVeto: 0,
    finalTallyResultYes: 0,
    id: 0,
    metadata: "",
    proposer: exports.defaultUser,
    status: "",
    submitTime: "",
    summary: "",
    title: "",
    totalDeposit: [exports.defaultToken],
    votingEndTime: "",
    votingStartTime: "",
};
exports.defaultGovDeposit = {
    amount: [exports.defaultToken],
    block: exports.defaultBlock,
    proposal: exports.defaultGovProposal,
    sender: exports.defaultUser,
};
exports.defaultGovVote = {
    block: exports.defaultBlock,
    option: "",
    proposal: exports.defaultGovProposal,
    sender: exports.defaultUser,
};
exports.defaultMarkPriceCandles = {
    close: 0,
    high: 0,
    low: 0,
    open: 0,
    volume: 0,
    volumeNotional: 0,
    pair: "",
    period: 0,
    periodInterval: "",
    periodStartTs: "",
    indexPriceTwapClose: 0,
};
exports.defaultPerpPositionChanges = {
    badDebt: exports.defaultToken,
    block: exports.defaultBlock,
    changeReason: "",
    eventSeqNo: 0,
    exchangedNotional: 0,
    exchangedSize: 0,
    fundingPayment: 0,
    latestCumulativePremiumFraction: 0,
    margin: 0,
    marginToUser: 0,
    openNotional: 0,
    pair: "",
    positionNotional: 0,
    realizedPnl: 0,
    size: 0,
    traderAddress: "",
    transactionFee: exports.defaultToken,
    txSeqNo: 0,
};
exports.defaultPerpLeaderboard = {
    avg_pct_pnl_rank: 0,
    avg_pct_pnl: 0,
    input_margin: 0,
    raw_pnl: 0,
    raw_pnl_with_unrealized: 0,
    trader_address: "",
};
exports.defaultGovernance = {
    govDeposits: [exports.defaultGovDeposit],
    govProposals: [exports.defaultGovProposal],
    govVotes: [exports.defaultGovVote],
};
exports.defaultDistributionCommission = {
    commission: [exports.defaultToken],
    validator: exports.defaultValidator,
};
exports.defaultDelegations = {
    amount: 0,
    delegator: exports.defaultUser,
    validator: exports.defaultValidator,
};
exports.defaultRedelegations = {
    amount: 0,
    delegator: exports.defaultUser,
    source_validator: exports.defaultValidator,
    destination_validator: exports.defaultValidator,
    completion_time: "",
    creation_block: exports.defaultBlock,
};
exports.defaultSpotLpPosition = {
    created_block: exports.defaultBlock,
    pool: exports.defaultPool,
    pool_shares: exports.defaultToken,
    user: exports.defaultUser,
};
exports.defaultSpotPoolSwap = {
    block: exports.defaultBlock,
    pool: exports.defaultPool,
    token_in: exports.defaultToken,
    token_out: exports.defaultToken,
    user: exports.defaultUser,
};
exports.defaultStatsFees = {
    feesLiquidations: 0,
    feesLiquidationsCumulative: 0,
    feesPerp: 0,
    feesPerpCumulative: 0,
    feesSwap: 0,
    feesSwapCumulative: 0,
    feesTotal: 0,
    feesTotalCumulative: 0,
    period: 0,
    periodInterval: "",
    periodStartTs: "",
};
exports.defaultPerpOpenInterest = {
    openInterestLong: 0,
    openInterestShort: 0,
    openInterestTotal: 0,
    period: 0,
    periodInterval: "",
    periodStartTs: "",
};
exports.defaultPerpPnl = {
    loss: 0,
    lossCumulative: 0,
    netPnl: 0,
    netPnlCumulative: 0,
    period: 0,
    periodInterval: "",
    periodStartTs: "",
    profit: 0,
    profitCumulative: 0,
};
exports.defaultTotals = {
    period: 0,
    periodInterval: "",
    periodStartTs: "",
    totalPerp: 0,
    totalFeesPerp: 0,
    totalFeesLiquidations: 0,
    totalOpenInterest: 0,
    totalTransactions: 0,
    totalSwap: 0,
    totalTvl: 0,
};
exports.defaultTvl = {
    period: 0,
    periodInterval: "",
    periodStartTs: "",
    tvlPerp: 0,
    tvlStablecoin: 0,
    tvlStaking: 0,
    tvlSwap: 0,
    tvlTotal: 0,
};
exports.defaultUsers = {
    newUsersLp: 0,
    newUsersLpCumulative: 0,
    newUsersPerp: 0,
    newUsersPerpCumulative: 0,
    newUsersSwap: 0,
    newUsersSwapCumulative: 0,
    newUsersTotal: 0,
    newUsersTotalCumulative: 0,
    period: 0,
    periodInterval: "",
    periodStartTs: "",
    userActionsPerp: 0,
    uniqueUsersLp: 0,
    uniqueUsersPerp: 0,
    uniqueUsersSwap: 0,
    uniqueUsersTotal: 0,
    userActionsLp: 0,
    userActionsSwap: 0,
    userActionsTotal: 0,
    newAuthUsers: 0,
    newAuthUsersCumulative: 0,
};
exports.defaultVolume = {
    volumePerp: 0,
    volumePerpCumulative: 0,
    volumeSwap: 0,
    volumeSwapCumulative: 0,
    volumeTotal: 0,
    volumeTotalCumulative: 0,
    period: 0,
    periodInterval: "",
    periodStartTs: "",
};
exports.defaultUnbondings = {
    amount: 0,
    completion_time: "",
    creation_block: exports.defaultBlock,
    delegator: exports.defaultUser,
    validator: exports.defaultValidator,
};
exports.defaultOraclePrice = {
    block: exports.defaultBlock,
    pair: "",
    eventSeqNo: 0,
    txSeqNo: 0,
    price: 0,
};
exports.defaultOracleEntry = {
    numVotes: 0,
    validator: exports.defaultValidator,
};
exports.defaultIbcChannel = {
    chain_id: "",
    chain_logo: "",
    chain_name: "",
    channel_id: "",
    counterparty_channel_id: "",
    revision_height: 0,
};
exports.defaultIbcTransfer = {
    amount: 0,
    memo: "",
    block: exports.defaultBlock,
    denom: "",
    sender: "",
    receiver: "",
};
exports.defaultIbc = {
    ibcChannels: {
        channels: [exports.defaultIbcChannel],
        revision_height: 0,
    },
    ibcTransfers: [exports.defaultIbcTransfer],
};
exports.defaultIbcChannelsResponse = {
    channels: [exports.defaultIbcChannel],
    revision_height: 0,
};
exports.defaultUserContract = {
    user: exports.defaultUser,
    contractAddress: "",
    contractType: "",
};
exports.defaultInflationDistribution = {
    block: exports.defaultBlock,
    communityPool: 0,
    eventSeqNo: 0,
    stakingRewards: 0,
    strategicReserve: 0,
    txSeqNo: 0,
};
exports.defaultInflationReward = {
    annualReward: 0,
    totalStaked: 0,
};
exports.defaultInflationInfo = {
    amount: 0,
    block: exports.defaultBlock,
    epochNumber: 0,
    epochProvisions: 0,
};
exports.defaultFeatureFlags = {
    gov: true,
    oracle: true,
    perp: true,
    spot: true,
    staking: true,
    wasm: true,
};
exports.defaultProxy = {
    bybit: {
        ask1Price: "",
        ask1Size: "",
        bid1Price: "",
        bid1Size: "",
        highPrice24h: "",
        lastPrice: "",
        lowPrice24h: "",
        prevPrice24h: "",
        price24hPcnt: "",
        symbol: "",
        turnover24h: "",
        volume24h: "",
    },
};
exports.defaultStakingHistoryItem = {
    action: "cancel",
    amount: 0,
    block: exports.defaultBlock,
    completion_time: "",
    delegator: exports.defaultUser,
    destination_validator: exports.defaultValidator,
    validator: exports.defaultValidator,
};
