// @nibiruchain/nibijs/dist/src/gql/utils/generated.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GQLValidatorStatus = exports.GQLValidatorOrder = exports.GQLUserOrder = exports.GQLUserContractsOrder = exports.GQLUnbondingOrder = exports.GQLStatsVolumeOrder = exports.GQLStatsUsersOrder = exports.GQLStatsTvlOrder = exports.GQLStatsTotalsOrder = exports.GQLStatsPerpPnlOrder = exports.GQLStatsPerpOpenInterestOrder = exports.GQLStatsInflationOrder = exports.GQLStatsFeesOrder = exports.GQLStakingHistoryOrder = exports.GQLStakingActionType = exports.GQLSpotPoolSwapOrder = exports.GQLSpotPoolOrder = exports.GQLSpotPoolJoinedOrder = exports.GQLSpotPoolExitedOrder = exports.GQLSpotPoolCreatedOrder = exports.GQLSpotLpPositionOrder = exports.GQLRedelegationOrder = exports.GQLPerpPositionOrder = exports.GQLPerpPositionChangeOrder = exports.GQLPerpMarketOrder = exports.GQLPerpLeaderboardOrder = exports.GQLOraclesOrder = exports.GQLOraclePricesOrder = exports.GQLMarkPriceCandlesOrder = exports.GQLInflationInfoOrder = exports.GQLInflationDistributionOrder = exports.GQLIbcTranferOrder = exports.GQLGovVotesOrder = exports.GQLGovProposalsOrder = exports.GQLGovDepositsOrder = exports.GQLDistributionCommissionOrder = exports.GQLDelegationOrder = exports.GQLContractEventsOrder = exports.GQLCommunityPoolOrder = void 0;
var GQLCommunityPoolOrder;
(function (GQLCommunityPoolOrder) {
    GQLCommunityPoolOrder["GQLDenom"] = "denom";
})(GQLCommunityPoolOrder = exports.GQLCommunityPoolOrder || (exports.GQLCommunityPoolOrder = {}));
var GQLContractEventsOrder;
(function (GQLContractEventsOrder) {
    GQLContractEventsOrder["GQLContractAddress"] = "contract_address";
    GQLContractEventsOrder["GQLSequence"] = "sequence";
    GQLContractEventsOrder["GQLType"] = "type";
})(GQLContractEventsOrder = exports.GQLContractEventsOrder || (exports.GQLContractEventsOrder = {}));
var GQLDelegationOrder;
(function (GQLDelegationOrder) {
    GQLDelegationOrder["GQLDelegatorAddress"] = "delegator_address";
    GQLDelegationOrder["GQLValidatorAddress"] = "validator_address";
})(GQLDelegationOrder = exports.GQLDelegationOrder || (exports.GQLDelegationOrder = {}));
var GQLDistributionCommissionOrder;
(function (GQLDistributionCommissionOrder) {
    GQLDistributionCommissionOrder["GQLValidatorAddress"] = "validator_address";
})(GQLDistributionCommissionOrder = exports.GQLDistributionCommissionOrder || (exports.GQLDistributionCommissionOrder = {}));
var GQLGovDepositsOrder;
(function (GQLGovDepositsOrder) {
    GQLGovDepositsOrder["GQLBlock"] = "block";
    GQLGovDepositsOrder["GQLProposalId"] = "proposal_id";
    GQLGovDepositsOrder["GQLSender"] = "sender";
})(GQLGovDepositsOrder = exports.GQLGovDepositsOrder || (exports.GQLGovDepositsOrder = {}));
var GQLGovProposalsOrder;
(function (GQLGovProposalsOrder) {
    GQLGovProposalsOrder["GQLDepositEndTime"] = "deposit_end_time";
    GQLGovProposalsOrder["GQLFinalTallyResultAbstain"] = "final_tally_result_abstain";
    GQLGovProposalsOrder["GQLFinalTallyResultNo"] = "final_tally_result_no";
    GQLGovProposalsOrder["GQLFinalTallyResultNoWithVeto"] = "final_tally_result_no_with_veto";
    GQLGovProposalsOrder["GQLFinalTallyResultYes"] = "final_tally_result_yes";
    GQLGovProposalsOrder["GQLId"] = "id";
    GQLGovProposalsOrder["GQLProposer"] = "proposer";
    GQLGovProposalsOrder["GQLStatus"] = "status";
    GQLGovProposalsOrder["GQLSubmitTime"] = "submit_time";
    GQLGovProposalsOrder["GQLSummary"] = "summary";
    GQLGovProposalsOrder["GQLTitle"] = "title";
    GQLGovProposalsOrder["GQLVotingEndTime"] = "voting_end_time";
    GQLGovProposalsOrder["GQLVotingStartTime"] = "voting_start_time";
})(GQLGovProposalsOrder = exports.GQLGovProposalsOrder || (exports.GQLGovProposalsOrder = {}));
var GQLGovVotesOrder;
(function (GQLGovVotesOrder) {
    GQLGovVotesOrder["GQLBlock"] = "block";
    GQLGovVotesOrder["GQLOption"] = "option";
    GQLGovVotesOrder["GQLProposalId"] = "proposal_id";
    GQLGovVotesOrder["GQLSender"] = "sender";
})(GQLGovVotesOrder = exports.GQLGovVotesOrder || (exports.GQLGovVotesOrder = {}));
var GQLIbcTranferOrder;
(function (GQLIbcTranferOrder) {
    GQLIbcTranferOrder["GQLBlock"] = "block";
    GQLIbcTranferOrder["GQLReceiver"] = "receiver";
    GQLIbcTranferOrder["GQLSender"] = "sender";
})(GQLIbcTranferOrder = exports.GQLIbcTranferOrder || (exports.GQLIbcTranferOrder = {}));
var GQLInflationDistributionOrder;
(function (GQLInflationDistributionOrder) {
    GQLInflationDistributionOrder["GQLSequence"] = "sequence";
})(GQLInflationDistributionOrder = exports.GQLInflationDistributionOrder || (exports.GQLInflationDistributionOrder = {}));
var GQLInflationInfoOrder;
(function (GQLInflationInfoOrder) {
    GQLInflationInfoOrder["GQLBlock"] = "block";
    GQLInflationInfoOrder["GQLEpochNumber"] = "epoch_number";
})(GQLInflationInfoOrder = exports.GQLInflationInfoOrder || (exports.GQLInflationInfoOrder = {}));
var GQLMarkPriceCandlesOrder;
(function (GQLMarkPriceCandlesOrder) {
    GQLMarkPriceCandlesOrder["GQLPeriod"] = "period";
    GQLMarkPriceCandlesOrder["GQLPeriodStartTs"] = "period_start_ts";
})(GQLMarkPriceCandlesOrder = exports.GQLMarkPriceCandlesOrder || (exports.GQLMarkPriceCandlesOrder = {}));
var GQLOraclePricesOrder;
(function (GQLOraclePricesOrder) {
    GQLOraclePricesOrder["GQLPair"] = "pair";
    GQLOraclePricesOrder["GQLPrice"] = "price";
    GQLOraclePricesOrder["GQLSequence"] = "sequence";
})(GQLOraclePricesOrder = exports.GQLOraclePricesOrder || (exports.GQLOraclePricesOrder = {}));
var GQLOraclesOrder;
(function (GQLOraclesOrder) {
    GQLOraclesOrder["GQLNumVotes"] = "num_votes";
    GQLOraclesOrder["GQLValidatorAddress"] = "validator_address";
})(GQLOraclesOrder = exports.GQLOraclesOrder || (exports.GQLOraclesOrder = {}));
var GQLPerpLeaderboardOrder;
(function (GQLPerpLeaderboardOrder) {
    GQLPerpLeaderboardOrder["GQLAvgPctPnl"] = "avg_pct_pnl";
    GQLPerpLeaderboardOrder["GQLAvgPctPnlRank"] = "avg_pct_pnl_rank";
    GQLPerpLeaderboardOrder["GQLRawPnl"] = "raw_pnl";
    GQLPerpLeaderboardOrder["GQLRawPnlWithUnrealized"] = "raw_pnl_with_unrealized";
    GQLPerpLeaderboardOrder["GQLTraderAddress"] = "trader_address";
})(GQLPerpLeaderboardOrder = exports.GQLPerpLeaderboardOrder || (exports.GQLPerpLeaderboardOrder = {}));
var GQLPerpMarketOrder;
(function (GQLPerpMarketOrder) {
    GQLPerpMarketOrder["GQLPair"] = "pair";
})(GQLPerpMarketOrder = exports.GQLPerpMarketOrder || (exports.GQLPerpMarketOrder = {}));
var GQLPerpPositionChangeOrder;
(function (GQLPerpPositionChangeOrder) {
    GQLPerpPositionChangeOrder["GQLExchangedNotional"] = "exchanged_notional";
    GQLPerpPositionChangeOrder["GQLExchangedSize"] = "exchanged_size";
    GQLPerpPositionChangeOrder["GQLMargin"] = "margin";
    GQLPerpPositionChangeOrder["GQLOpenNotional"] = "open_notional";
    GQLPerpPositionChangeOrder["GQLPair"] = "pair";
    GQLPerpPositionChangeOrder["GQLPositionNotional"] = "position_notional";
    GQLPerpPositionChangeOrder["GQLSequence"] = "sequence";
    GQLPerpPositionChangeOrder["GQLSize"] = "size";
})(GQLPerpPositionChangeOrder = exports.GQLPerpPositionChangeOrder || (exports.GQLPerpPositionChangeOrder = {}));
var GQLPerpPositionOrder;
(function (GQLPerpPositionOrder) {
    GQLPerpPositionOrder["GQLCreatedBlock"] = "created_block";
    GQLPerpPositionOrder["GQLPair"] = "pair";
    GQLPerpPositionOrder["GQLTraderAddress"] = "trader_address";
})(GQLPerpPositionOrder = exports.GQLPerpPositionOrder || (exports.GQLPerpPositionOrder = {}));
var GQLRedelegationOrder;
(function (GQLRedelegationOrder) {
    GQLRedelegationOrder["GQLCompletionTime"] = "completion_time";
    GQLRedelegationOrder["GQLCreationHeight"] = "creation_height";
    GQLRedelegationOrder["GQLDelegatorAddress"] = "delegator_address";
    GQLRedelegationOrder["GQLDestinationValidatorAddress"] = "destination_validator_address";
    GQLRedelegationOrder["GQLSourceValidatorAddress"] = "source_validator_address";
})(GQLRedelegationOrder = exports.GQLRedelegationOrder || (exports.GQLRedelegationOrder = {}));
var GQLSpotLpPositionOrder;
(function (GQLSpotLpPositionOrder) {
    GQLSpotLpPositionOrder["GQLPoolId"] = "pool_id";
    GQLSpotLpPositionOrder["GQLUserAddress"] = "user_address";
})(GQLSpotLpPositionOrder = exports.GQLSpotLpPositionOrder || (exports.GQLSpotLpPositionOrder = {}));
var GQLSpotPoolCreatedOrder;
(function (GQLSpotPoolCreatedOrder) {
    GQLSpotPoolCreatedOrder["GQLPoolId"] = "pool_id";
    GQLSpotPoolCreatedOrder["GQLUserAddress"] = "user_address";
})(GQLSpotPoolCreatedOrder = exports.GQLSpotPoolCreatedOrder || (exports.GQLSpotPoolCreatedOrder = {}));
var GQLSpotPoolExitedOrder;
(function (GQLSpotPoolExitedOrder) {
    GQLSpotPoolExitedOrder["GQLPoolId"] = "pool_id";
    GQLSpotPoolExitedOrder["GQLUserAddress"] = "user_address";
})(GQLSpotPoolExitedOrder = exports.GQLSpotPoolExitedOrder || (exports.GQLSpotPoolExitedOrder = {}));
var GQLSpotPoolJoinedOrder;
(function (GQLSpotPoolJoinedOrder) {
    GQLSpotPoolJoinedOrder["GQLPoolId"] = "pool_id";
    GQLSpotPoolJoinedOrder["GQLUserAddress"] = "user_address";
})(GQLSpotPoolJoinedOrder = exports.GQLSpotPoolJoinedOrder || (exports.GQLSpotPoolJoinedOrder = {}));
var GQLSpotPoolOrder;
(function (GQLSpotPoolOrder) {
    GQLSpotPoolOrder["GQLPoolId"] = "pool_id";
})(GQLSpotPoolOrder = exports.GQLSpotPoolOrder || (exports.GQLSpotPoolOrder = {}));
var GQLSpotPoolSwapOrder;
(function (GQLSpotPoolSwapOrder) {
    GQLSpotPoolSwapOrder["GQLBlock"] = "block";
    GQLSpotPoolSwapOrder["GQLPoolId"] = "pool_id";
    GQLSpotPoolSwapOrder["GQLUserAddress"] = "user_address";
})(GQLSpotPoolSwapOrder = exports.GQLSpotPoolSwapOrder || (exports.GQLSpotPoolSwapOrder = {}));
var GQLStakingActionType;
(function (GQLStakingActionType) {
    GQLStakingActionType["GQLCancel"] = "cancel";
    GQLStakingActionType["GQLDelegate"] = "delegate";
    GQLStakingActionType["GQLRedelegate"] = "redelegate";
    GQLStakingActionType["GQLRedelegateCompleted"] = "redelegate_completed";
    GQLStakingActionType["GQLUnbond"] = "unbond";
    GQLStakingActionType["GQLUnbondCompleted"] = "unbond_completed";
    GQLStakingActionType["GQLWithdraw"] = "withdraw";
})(GQLStakingActionType = exports.GQLStakingActionType || (exports.GQLStakingActionType = {}));
var GQLStakingHistoryOrder;
(function (GQLStakingHistoryOrder) {
    GQLStakingHistoryOrder["GQLAction"] = "action";
    GQLStakingHistoryOrder["GQLAmount"] = "amount";
    GQLStakingHistoryOrder["GQLDelegator"] = "delegator";
    GQLStakingHistoryOrder["GQLSequence"] = "sequence";
    GQLStakingHistoryOrder["GQLValidator"] = "validator";
})(GQLStakingHistoryOrder = exports.GQLStakingHistoryOrder || (exports.GQLStakingHistoryOrder = {}));
var GQLStatsFeesOrder;
(function (GQLStatsFeesOrder) {
    GQLStatsFeesOrder["GQLFeesLiquidations"] = "fees_liquidations";
    GQLStatsFeesOrder["GQLFeesLiquidationsCumulative"] = "fees_liquidations_cumulative";
    GQLStatsFeesOrder["GQLFeesPerp"] = "fees_perp";
    GQLStatsFeesOrder["GQLFeesPerpCumulative"] = "fees_perp_cumulative";
    GQLStatsFeesOrder["GQLFeesSwap"] = "fees_swap";
    GQLStatsFeesOrder["GQLFeesSwapCumulative"] = "fees_swap_cumulative";
    GQLStatsFeesOrder["GQLFeesTotal"] = "fees_total";
    GQLStatsFeesOrder["GQLFeesTotalCumulative"] = "fees_total_cumulative";
    GQLStatsFeesOrder["GQLPeriod"] = "period";
    GQLStatsFeesOrder["GQLPeriodStartTs"] = "period_start_ts";
})(GQLStatsFeesOrder = exports.GQLStatsFeesOrder || (exports.GQLStatsFeesOrder = {}));
var GQLStatsInflationOrder;
(function (GQLStatsInflationOrder) {
    GQLStatsInflationOrder["GQLCommunityPool"] = "community_pool";
    GQLStatsInflationOrder["GQLCommunityPoolCumulative"] = "community_pool_cumulative";
    GQLStatsInflationOrder["GQLPeriod"] = "period";
    GQLStatsInflationOrder["GQLPeriodStartTs"] = "period_start_ts";
    GQLStatsInflationOrder["GQLStaking"] = "staking";
    GQLStatsInflationOrder["GQLStakingCumulative"] = "staking_cumulative";
    GQLStatsInflationOrder["GQLStrategicReserve"] = "strategic_reserve";
    GQLStatsInflationOrder["GQLStrategicReserveCumulative"] = "strategic_reserve_cumulative";
})(GQLStatsInflationOrder = exports.GQLStatsInflationOrder || (exports.GQLStatsInflationOrder = {}));
var GQLStatsPerpOpenInterestOrder;
(function (GQLStatsPerpOpenInterestOrder) {
    GQLStatsPerpOpenInterestOrder["GQLOpenInterestLong"] = "open_interest_long";
    GQLStatsPerpOpenInterestOrder["GQLOpenInterestShort"] = "open_interest_short";
    GQLStatsPerpOpenInterestOrder["GQLOpenInterestTotal"] = "open_interest_total";
    GQLStatsPerpOpenInterestOrder["GQLPeriod"] = "period";
    GQLStatsPerpOpenInterestOrder["GQLPeriodStartTs"] = "period_start_ts";
})(GQLStatsPerpOpenInterestOrder = exports.GQLStatsPerpOpenInterestOrder || (exports.GQLStatsPerpOpenInterestOrder = {}));
var GQLStatsPerpPnlOrder;
(function (GQLStatsPerpPnlOrder) {
    GQLStatsPerpPnlOrder["GQLLoss"] = "loss";
    GQLStatsPerpPnlOrder["GQLLossCumulative"] = "loss_cumulative";
    GQLStatsPerpPnlOrder["GQLNetPnl"] = "net_pnl";
    GQLStatsPerpPnlOrder["GQLNetPnlCumulative"] = "net_pnl_cumulative";
    GQLStatsPerpPnlOrder["GQLPeriod"] = "period";
    GQLStatsPerpPnlOrder["GQLPeriodStartTs"] = "period_start_ts";
    GQLStatsPerpPnlOrder["GQLProfit"] = "profit";
    GQLStatsPerpPnlOrder["GQLProfitCumulative"] = "profit_cumulative";
})(GQLStatsPerpPnlOrder = exports.GQLStatsPerpPnlOrder || (exports.GQLStatsPerpPnlOrder = {}));
var GQLStatsTotalsOrder;
(function (GQLStatsTotalsOrder) {
    GQLStatsTotalsOrder["GQLPeriod"] = "period";
    GQLStatsTotalsOrder["GQLPeriodStartTs"] = "period_start_ts";
    GQLStatsTotalsOrder["GQLTotalFeesLiquidations"] = "total_fees_liquidations";
    GQLStatsTotalsOrder["GQLTotalFeesPerp"] = "total_fees_perp";
    GQLStatsTotalsOrder["GQLTotalOpenInterest"] = "total_open_interest";
    GQLStatsTotalsOrder["GQLTotalPerp"] = "total_perp";
    GQLStatsTotalsOrder["GQLTotalSwap"] = "total_swap";
    GQLStatsTotalsOrder["GQLTotalTransactions"] = "total_transactions";
    GQLStatsTotalsOrder["GQLTotalTvl"] = "total_tvl";
})(GQLStatsTotalsOrder = exports.GQLStatsTotalsOrder || (exports.GQLStatsTotalsOrder = {}));
var GQLStatsTvlOrder;
(function (GQLStatsTvlOrder) {
    GQLStatsTvlOrder["GQLPeriod"] = "period";
    GQLStatsTvlOrder["GQLPeriodStartTs"] = "period_start_ts";
    GQLStatsTvlOrder["GQLTvlPerp"] = "tvl_perp";
    GQLStatsTvlOrder["GQLTvlStablecoin"] = "tvl_stablecoin";
    GQLStatsTvlOrder["GQLTvlStaking"] = "tvl_staking";
    GQLStatsTvlOrder["GQLTvlSwap"] = "tvl_swap";
    GQLStatsTvlOrder["GQLTvlTotal"] = "tvl_total";
})(GQLStatsTvlOrder = exports.GQLStatsTvlOrder || (exports.GQLStatsTvlOrder = {}));
var GQLStatsUsersOrder;
(function (GQLStatsUsersOrder) {
    GQLStatsUsersOrder["GQLNewAuthUsers"] = "new_auth_users";
    GQLStatsUsersOrder["GQLNewAuthUsersCumulative"] = "new_auth_users_cumulative";
    GQLStatsUsersOrder["GQLNewUsersLp"] = "new_users_lp";
    GQLStatsUsersOrder["GQLNewUsersLpCumulative"] = "new_users_lp_cumulative";
    GQLStatsUsersOrder["GQLNewUsersPerp"] = "new_users_perp";
    GQLStatsUsersOrder["GQLNewUsersPerpCumulative"] = "new_users_perp_cumulative";
    GQLStatsUsersOrder["GQLNewUsersSwap"] = "new_users_swap";
    GQLStatsUsersOrder["GQLNewUsersSwapCumulative"] = "new_users_swap_cumulative";
    GQLStatsUsersOrder["GQLNewUsersTotal"] = "new_users_total";
    GQLStatsUsersOrder["GQLNewUsersTotalCumulative"] = "new_users_total_cumulative";
    GQLStatsUsersOrder["GQLPeriod"] = "period";
    GQLStatsUsersOrder["GQLPeriodStartTs"] = "period_start_ts";
    GQLStatsUsersOrder["GQLUniqueUsersLp"] = "unique_users_lp";
    GQLStatsUsersOrder["GQLUniqueUsersPerp"] = "unique_users_perp";
    GQLStatsUsersOrder["GQLUniqueUsersSwap"] = "unique_users_swap";
    GQLStatsUsersOrder["GQLUniqueUsersTotal"] = "unique_users_total";
    GQLStatsUsersOrder["GQLUserActionsLp"] = "user_actions_lp";
    GQLStatsUsersOrder["GQLUserActionsPerp"] = "user_actions_perp";
    GQLStatsUsersOrder["GQLUserActionsSwap"] = "user_actions_swap";
    GQLStatsUsersOrder["GQLUserActionsTotal"] = "user_actions_total";
})(GQLStatsUsersOrder = exports.GQLStatsUsersOrder || (exports.GQLStatsUsersOrder = {}));
var GQLStatsVolumeOrder;
(function (GQLStatsVolumeOrder) {
    GQLStatsVolumeOrder["GQLPeriod"] = "period";
    GQLStatsVolumeOrder["GQLPeriodStartTs"] = "period_start_ts";
    GQLStatsVolumeOrder["GQLVolumePerp"] = "volume_perp";
    GQLStatsVolumeOrder["GQLVolumePerpCumulative"] = "volume_perp_cumulative";
    GQLStatsVolumeOrder["GQLVolumeSwap"] = "volume_swap";
    GQLStatsVolumeOrder["GQLVolumeSwapCumulative"] = "volume_swap_cumulative";
    GQLStatsVolumeOrder["GQLVolumeTotal"] = "volume_total";
    GQLStatsVolumeOrder["GQLVolumeTotalCumulative"] = "volume_total_cumulative";
})(GQLStatsVolumeOrder = exports.GQLStatsVolumeOrder || (exports.GQLStatsVolumeOrder = {}));
var GQLUnbondingOrder;
(function (GQLUnbondingOrder) {
    GQLUnbondingOrder["GQLCompletionTime"] = "completion_time";
    GQLUnbondingOrder["GQLCreationHeight"] = "creation_height";
    GQLUnbondingOrder["GQLDelegatorAddress"] = "delegator_address";
    GQLUnbondingOrder["GQLValidatorAddress"] = "validator_address";
})(GQLUnbondingOrder = exports.GQLUnbondingOrder || (exports.GQLUnbondingOrder = {}));
var GQLUserContractsOrder;
(function (GQLUserContractsOrder) {
    GQLUserContractsOrder["GQLContractAddress"] = "contract_address";
    GQLUserContractsOrder["GQLContractType"] = "contract_type";
    GQLUserContractsOrder["GQLUserAddress"] = "user_address";
})(GQLUserContractsOrder = exports.GQLUserContractsOrder || (exports.GQLUserContractsOrder = {}));
var GQLUserOrder;
(function (GQLUserOrder) {
    GQLUserOrder["GQLAddress"] = "address";
    GQLUserOrder["GQLCreatedBlock"] = "created_block";
})(GQLUserOrder = exports.GQLUserOrder || (exports.GQLUserOrder = {}));
var GQLValidatorOrder;
(function (GQLValidatorOrder) {
    GQLValidatorOrder["GQLJailed"] = "jailed";
    GQLValidatorOrder["GQLMoniker"] = "moniker";
    GQLValidatorOrder["GQLOperatorAddress"] = "operator_address";
    GQLValidatorOrder["GQLStatus"] = "status";
    GQLValidatorOrder["GQLTokens"] = "tokens";
})(GQLValidatorOrder = exports.GQLValidatorOrder || (exports.GQLValidatorOrder = {}));
var GQLValidatorStatus;
(function (GQLValidatorStatus) {
    GQLValidatorStatus["GQLBonded"] = "BONDED";
    GQLValidatorStatus["GQLUnbonded"] = "UNBONDED";
    GQLValidatorStatus["GQLUnbonding"] = "UNBONDING";
})(GQLValidatorStatus = exports.GQLValidatorStatus || (exports.GQLValidatorStatus = {}));
