// @nibiruchain/nibijs/dist/src/protojs/nibiru/devgas/v1/event.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventPayoutDevGas = exports.EventUpdateDevGas = exports.EventCancelDevGas = exports.EventRegisterDevGas = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
function createBaseEventRegisterDevGas() {
    return { deployer: "", contract: "", withdrawer: "" };
}
exports.EventRegisterDevGas = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.deployer !== "") {
            writer.uint32(10).string(message.deployer);
        }
        if (message.contract !== "") {
            writer.uint32(18).string(message.contract);
        }
        if (message.withdrawer !== "") {
            writer.uint32(26).string(message.withdrawer);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventRegisterDevGas();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.deployer = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.contract = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.withdrawer = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            deployer: isSet(object.deployer) ? String(object.deployer) : "",
            contract: isSet(object.contract) ? String(object.contract) : "",
            withdrawer: isSet(object.withdrawer) ? String(object.withdrawer) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.deployer !== undefined && (obj.deployer = message.deployer);
        message.contract !== undefined && (obj.contract = message.contract);
        message.withdrawer !== undefined && (obj.withdrawer = message.withdrawer);
        return obj;
    },
    create(base) {
        return exports.EventRegisterDevGas.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseEventRegisterDevGas();
        message.deployer = (_a = object.deployer) !== null && _a !== void 0 ? _a : "";
        message.contract = (_b = object.contract) !== null && _b !== void 0 ? _b : "";
        message.withdrawer = (_c = object.withdrawer) !== null && _c !== void 0 ? _c : "";
        return message;
    },
};
function createBaseEventCancelDevGas() {
    return { deployer: "", contract: "" };
}
exports.EventCancelDevGas = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.deployer !== "") {
            writer.uint32(10).string(message.deployer);
        }
        if (message.contract !== "") {
            writer.uint32(18).string(message.contract);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventCancelDevGas();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.deployer = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.contract = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            deployer: isSet(object.deployer) ? String(object.deployer) : "",
            contract: isSet(object.contract) ? String(object.contract) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.deployer !== undefined && (obj.deployer = message.deployer);
        message.contract !== undefined && (obj.contract = message.contract);
        return obj;
    },
    create(base) {
        return exports.EventCancelDevGas.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseEventCancelDevGas();
        message.deployer = (_a = object.deployer) !== null && _a !== void 0 ? _a : "";
        message.contract = (_b = object.contract) !== null && _b !== void 0 ? _b : "";
        return message;
    },
};
function createBaseEventUpdateDevGas() {
    return { deployer: "", contract: "", withdrawer: "" };
}
exports.EventUpdateDevGas = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.deployer !== "") {
            writer.uint32(10).string(message.deployer);
        }
        if (message.contract !== "") {
            writer.uint32(18).string(message.contract);
        }
        if (message.withdrawer !== "") {
            writer.uint32(26).string(message.withdrawer);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventUpdateDevGas();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.deployer = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.contract = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.withdrawer = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            deployer: isSet(object.deployer) ? String(object.deployer) : "",
            contract: isSet(object.contract) ? String(object.contract) : "",
            withdrawer: isSet(object.withdrawer) ? String(object.withdrawer) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.deployer !== undefined && (obj.deployer = message.deployer);
        message.contract !== undefined && (obj.contract = message.contract);
        message.withdrawer !== undefined && (obj.withdrawer = message.withdrawer);
        return obj;
    },
    create(base) {
        return exports.EventUpdateDevGas.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseEventUpdateDevGas();
        message.deployer = (_a = object.deployer) !== null && _a !== void 0 ? _a : "";
        message.contract = (_b = object.contract) !== null && _b !== void 0 ? _b : "";
        message.withdrawer = (_c = object.withdrawer) !== null && _c !== void 0 ? _c : "";
        return message;
    },
};
function createBaseEventPayoutDevGas() {
    return { payouts: "" };
}
exports.EventPayoutDevGas = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.payouts !== "") {
            writer.uint32(10).string(message.payouts);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventPayoutDevGas();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payouts = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { payouts: isSet(object.payouts) ? String(object.payouts) : "" };
    },
    toJSON(message) {
        const obj = {};
        message.payouts !== undefined && (obj.payouts = message.payouts);
        return obj;
    },
    create(base) {
        return exports.EventPayoutDevGas.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseEventPayoutDevGas();
        message.payouts = (_a = object.payouts) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
