// @nibiruchain/nibijs/dist/src/protojs/eth/types/v1/dynamic_fee.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionOptionDynamicFeeTx = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
function createBaseExtensionOptionDynamicFeeTx() {
    return { maxPriorityPrice: "" };
}
exports.ExtensionOptionDynamicFeeTx = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.maxPriorityPrice !== "") {
            writer.uint32(10).string(message.maxPriorityPrice);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseExtensionOptionDynamicFeeTx();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.maxPriorityPrice = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { maxPriorityPrice: isSet(object.maxPriorityPrice) ? String(object.maxPriorityPrice) : "" };
    },
    toJSON(message) {
        const obj = {};
        message.maxPriorityPrice !== undefined && (obj.maxPriorityPrice = message.maxPriorityPrice);
        return obj;
    },
    create(base) {
        return exports.ExtensionOptionDynamicFeeTx.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseExtensionOptionDynamicFeeTx();
        message.maxPriorityPrice = (_a = object.maxPriorityPrice) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
